### DISCUSSION TITLE
Getting 401 issue when file opener plugin installation

### DISCUSSION THREAD

**Discussion by anusha-kaparapu** [capawesome-team/capacitor-plugins#161] [General]
Hi @robingenz we got the below issue when installing the file opener plugin. File opener is a paid plugin?

Could you please provide the info.

**npm install @capawesome-team/capacitor-file-opener
npm ERR! code E401
npm ERR! 401 Unauthorized - GET https://npm.pkg.github.com/@capawesome-team%2fcapacitor-file-opener - unauthenticated: User cannot be authenticated with the token provided.**

if we use capacitor-file-picker plugin in a project, is there any impact on the future?

**Comment by robingenz** [comment]
No, it's also available via the public npm registry. If you want to use the private GitHub npm registry, you need to be a [Capawesome Insider](https://capawesome.io/sponsors/insiders/).

**Reply by anusha-kaparapu** [reply]
Thanks @robingenz , I have two more questions, please help me for the below questions.

1. Could you please confirm that capacitor-file-picker won't switch to a paid version in the future? 
2. When will the file-opener plugin be made available to the general public?

**Reply by robingenz** [reply]
1. We do not plan to change the license for the Capacitor File Picker plugin.
2. It's already public, see [here](https://www.npmjs.com/package/@capawesome-team/capacitor-file-opener).

**Reply by anusha-kaparapu** [reply]
Thanks @robingenz, 
We used the One of the API **'https://Xxxxx/View/upload-file.png?contentType=image/png'** to get the actual image or file from Api response, but for image we are getting as a preview, for files like xls, pdf etc.. getting the response as blob.
Please help me how to handle this scenario? Thanks in Advance.
For example:
open = async (file) => {
    await FileOpener.openFile({
      path: 'Xxxxx/View/upload-file.png?contentType=image/png',
    });
  };
**Got this error**
Failed to open URL : Error Domain=NSOSStatusErrorDomain Code=-50 "invalid input parameters" UserInfo={NSDebugDescription=invalid input parameters, _LSLine=249, _LSFunction=-[_LSDOpenClient openURL:fileHandle:options:completionHandler:]}

**Reply by robingenz** [reply]
You need to [download the file](https://capawesome.io/blog/the-file-handling-guide-for-capacitor/#download-a-file) first before opening it using the File Opener plugin.
